README.md

```diff
@@ -13,175 +13,413 @@ Compartmentalization of host behavior hooks for JS.
+    instance: string,
+  }
+
+  // To create an alias to an existing module instance given the full specifier
+  // of the module in a different compartment:
+  | {
+    instance: string,
+
+    // A compartment instance.
+    // We do not preclude the possibility that compartment is the same compartment,
+    // as that is possible to achieve in moduleMapHook and loadHook, albeit not
+    // in moduleMap.
+    compartment: Compartment,
+  }
+
+  // To create an alias to an existing module instance given a module exports
+  // namespace object that can pass a brand check:
+  | {
+    namespace: ModuleExportsNamespace,
+  }
+
+  // If the given namespace does not pass a ModuleExportsNamespace brandcheck,
+  // the compartment will not support live bindings to the properties of that
+  // object, but will instead produce an emulation of a module exports namespace,
+  // using a frozen snapshot of the own properties of the given object.
+  // The module exports namespace for this module does not reflect any future
+  // changes to the shape of the given object.
+  | {
+    namespace: ^ModuleExportsNamespace,
+  };
+
 type CompartmentConstructorOptions = {
-  // JF has a better way for:
-  //   randomHook(): number; // Use for Math.random()
-  //   nowHook(): number; // Use for both Date.now() and new Date(),
-  // Supplied during Module instance creation instead of option
-  //   hasSourceTextAvailableHook(scriptOrModule): bool; // Used for censorship
-  resolveHook(name: string, referrer: FullSpecifier): FullSpecifier
-
-  // timing
-  // importHook delegates to importNowHook FIRST,
-  // they share a memo cache
-  // order to ensures importNow never allows async value of import
-  // to be accessed prior to any attached promise
-  importHook(fullSpec: FullSpecifier): Promise<StaticModuleRecord>;
-  importNowHook(fullSpec: FullSpecifier): StaticModuleRecord?;
-
-  // copy own props after return
-  importMetaHook(fullSpec: FullSpecifier): object
-
-  // e.g.: 'fr-FR' - Affects appropriate ECMA-402 APIs within Compartment
-  localeHook(): string;
-  // This is important to be able to override for deterministic testing and such
-  localTZAHook(): string;
-
-  // determines if the fn is acting as an "eval" function
-  isDirectEvalHook(evalFunctionRef: any): boolean;
-
-  // prep for trusted types non-string
-  canCompileHook(source: any, {
-    evaluator: functionRef, // can be a value from isDirectEvalHook
-    isDirect?: boolean
-  }): boolean; // need to allow mimicing CSP including nonces
+  // inherit, when true, indicates that this compartment shares the globals, global
+  // contour, and evaluators of the surrounding compartment.
+  // This would interact with the SES `lockdown` function: after lockdown,
+  // a true would be incoherent and throw an exception.
+  // When false, the compartment's globalThis would be populated only
+  // with a subset of the language's intrinsics and the compartment's
```

Comment:
Definitely not. I can’t imagine us permitting any object not currently in the SES-shim Lockdown permits by default. In order to avoid arguing about the exact list, my hope is that the only folks who care about `borrowGlobals: false` are those who in turn intend to also use Lockdown. `borrowGlobals: true` avoids that conversation in the cases where people are likely to object to the exact list of intrinsics in a `borrowGlobals: false` compartment.